src/bvals/comms/boundary_communication.cpp

```diff
@@ -286,48 +320,57 @@ TaskStatus SetBounds(std::shared_ptr<MeshData<Real>> &md) {
-            Kokkos::parallel_for(
-                Kokkos::TeamThreadRange<>(team_member, idxer.size() / Ni),
-                [&](const int idx) {
-                  const auto [t, u, v, k, j, i] = idxer(idx * Ni);
-                  const int tt = t;
-                  const int uu = u;
-                  const int vv = v;
-                  const int kk = k;
-                  const int jj = j;
-                  const int ii = i;
-                  Kokkos::parallel_for(
-                      Kokkos::ThreadVectorRange<>(team_member, Ni), [&](int m) {
-                        const auto [il, jl, kl] =
-                            lcoord_trans.InverseTransform({ii + m, jj, kk});
-                        if (idxer.IsActive(kl, jl, il))
-                          var(iel, tt, uu, vv, kl, jl, il) = default_val;
-                      });
-                });
+          if (bnd_info(b).allocated) {
+            const int n_units = idxer.size() / Ni;
+            SplitFlatIndexRangeAmongTeams split(nteams_per_buffer, work_chunk_size,
+                                                n_units);
+            const auto [start, end] = split.GetIdxRange(bteam);
+            if (start >= end) {
+              idx_offset += idxer.size();
+              continue;
+            }
+
+            if (bnd_info(b).buf_allocated) {
+              Kokkos::parallel_for(
+                  Kokkos::TeamThreadRange<>(team_member, start, end), [&](const int idx) {
+                    Real *buf = &bnd_info(b).buf(idx * Ni + idx_offset);
+                    const auto [t, u, v, k, j, i] = idxer(idx * Ni);
+                    // Have to do this because of some weird issue about structure
+                    // bindings being captured
+                    const int tt = t;
+                    const int uu = u;
+                    const int vv = v;
+                    const int kk = k;
+                    const int jj = j;
+                    const int ii = i;
+                    Kokkos::parallel_for(
+                        Kokkos::ThreadVectorRange<>(team_member, Ni), [&](int m) {
+                          const auto [il, jl, kl] =
+                              lcoord_trans.InverseTransform({ii + m, jj, kk});
+                          if (idxer.IsActive(kl, jl, il))
+                            var(iel, tt, uu, vv, kl, jl, il) = fac * buf[m];
+                        });
+                  });
+            } else if (bound_type != BoundaryType::flxcor_recv) {
+              const Real default_val = bnd_info(b).var.sparse_default_val;
+              Kokkos::parallel_for(
+                  Kokkos::TeamThreadRange<>(team_member, start, end), [&](const int idx) {
+                    const auto [t, u, v, k, j, i] = idxer(idx * Ni);
+                    const int tt = t;
+                    const int uu = u;
+                    const int vv = v;
+                    const int kk = k;
+                    const int jj = j;
+                    const int ii = i;
```

Comment:
structured binding working properly with when 😢

Reply:
This pains me greatly...